controller/volume_controller.go

```diff
@@ -838,10 +838,17 @@ func (vc *VolumeController) ReconcileVolumeState(v *longhorn.Volume, e *longhorn
 
 		//Automatically detach the new restored volume, but not the standby volume
 		if v.Spec.RestorationRequired == true && v.Spec.FromBackup != "" && v.Status.State == types.VolumeStateAttached {
-			if v.Spec.Standby == false {
-				v.Spec.NodeID = ""
+			if v.Spec.Standby {
```

Comment:
Need a comment on how those flags suppose to work. The interaction, detach/attach, standby vs normal etc.